ern-orchestrator/src/codepush.ts

```diff
@@ -140,19 +142,30 @@ export async function performCodePushPromote(
         PackagePath.fromString(jsapiimpl)
       )
 
-      const nativeDependenciesVersionAligned = await compatibility.areCompatible(
-        miniApps,
-        targetNapDescriptor
-      )
-
-      if (!nativeDependenciesVersionAligned && force) {
-        log.warn(
-          'Native dependencies versions are not aligned but ignoring due to the use of force flag'
-        )
-      } else if (!nativeDependenciesVersionAligned && !force) {
-        throw new Error(
-          'Native dependencies versions of MiniApps are not aligned. Relaunch the operation with the force flag if you wish to ignore.'
+      // If sourceNapDescriptor.version === targetNapDescriptor.version skip the compatibility check
+      // As the compatibility check would have already happened as part of the code-push release
+      // If skipNativeDependenciesVersionAlignedCheck is set to true
+      // and sourceNapDescriptor and targetNapDescriptor versions are different, explicity skip the check
+      // would be required for patch version promotion
+      if (
+        sourceNapDescriptor &&
+        sourceNapDescriptor.version !== targetNapDescriptor.version &&
```

Comment:
Just confirming: It _is_ possible for `sourceNapDescriptor` to be null, but _not_ for `targetNapDescriptor`?

Reply:
@friederbluemle https://github.com/electrode-io/electrode-native/pull/1574/files#diff-fa17892fa392626ed2f1932956acba0cR123
The check is already here. It would be okay to skip `sourceNapDescriptor` check, though like being defensive :)